api/envoy/config/filter/network/http_connection_manager/v2/http_connection_manager.proto

```diff
@@ -287,6 +287,12 @@ message HttpConnectionManager {
   // is not desired it can be disabled.
   google.protobuf.BoolValue generate_request_id = 15;
 
+  // Whether the connection manager will keep the :ref:`x-request-id
+  // <config_http_conn_man_headers_x-request-id>` header if passed for a request that is edge
+  // (Edge request is the request from external clients to front Envoy) and not reset it, which
+  // is the current Envoy behaviour. This defaults to false.
+  bool edge_accept_request_id = 32;
```

Comment:
Prefer a verbed name, e.g. `preserve_requested_id_at_edge`. However, I think talking about edge here is confusing, we use internal/external elsewhere, e.g. we talk about "internal" and "trusted clients" at https://www.envoyproxy.io/docs/envoy/latest/configuration/http_conn_man/headers#config-http-conn-man-headers-x-forwarded-for, ideally we stick with consistent terminology here.
/wait

Reply:
should preserver_external_request_id work?

Reply:
SGTM

Reply:
done